### DISCUSSION TITLE
Running multiple singularity exec commands in Bash script

### DISCUSSION THREAD

**Discussion by treigerm** [sylabs/singularity#2815] [Q&A]
I am trying to create a Bash script which executes a command in a Singularity container multiple times. However, I get an odd behaviour insofar that the script terminates with an error after I have run the first `singularity exec` command.

Example script exhibiting the behaviour:
```
#!/usr/bin/env bash

singularity exec scam6_ppe.sif /home/scam/run_scam6_configuration.sh
echo $?
echo "first success"
singularity exec scam6_ppe.sif /home/scam/run_scam6_configuration.sh
echo $?
echo "second success"
```
The script succesfully executes the first `singularity exec` command and returns with exit code 0 but when it reaches the second `singularity exec` command it errors with: `[23]  + 3455 suspended (tty output)  ./scripts/scam_evals.sh`.

I have already tried setting `stty -tostop` as suggested [here](https://stackoverflow.com/questions/24056102/why-do-i-get-suspended-tty-output-in-one-terminal-but-not-in-others) in my shell but that did not resolve my issue.

I am able to succesfully run multiple `singularity exec` commands outside a script using the command line. Sadly, I cannot provide a MWE for the `run_scam6_configuration.sh` script as it calls a complex simulator written in Fortran. However, the `run_scam6_configuration.sh` script itself does not start any background processes if that is of help.

Any ideas what might be causing the issue and how I should go about debugging it?

Singularity version: 3.0.1
OS: Ubuntu 16.04.7 LTS (GNU/Linux 4.15.0-220-generic x86_64)
Shell: zsh 5.1.1 (x86_64-ubuntu-linux-gnu

**Comment by dtrudg** [comment]
The Singularity and Ubuntu versions used here are very old, and not supported. I'd recommend trying to run the same script on a newer OS, using a current Singularity version (4.1.2). It's unlikely anyone will have an old install to try and reproduce this on.

**Comment by treigerm** [comment]
Thank you for the response. Unfortunately, this is happening on a shared resource for which I have no admin rights. I will try to get the admin to provide updates.
